SNIPS/snip-16.md

```diff
@@ -0,0 +1,46 @@
+# Draft---
+snip: 16
+title: Deprecation of transaction versions 0,1,2
+author: Ilia Volokh <iliav@starkware.co>
+discussions-to: 
+status: Draft
+type: Core
+created: July 2, 2024
+---
+
+## Simple Summary
+In the context of a fee market, transaction versions 0,1,2 are inferior to and incompatible with transaction version 3: they support a different fee token and have inefrior bid structure.
+
+We propose to depracate transaction versions 0,1,2 and to stop supporting them once the mempool is implemented in the sequencer.
+
+## Motivation
+
+The sequencer currently processes transactions in FIFO. During congestion block space becomes scarce. To improve UX in such cases, Starknet will have a fee market which will allow users to express value and time preference.
+
+Transaction version 3 was introduced with a 1559-type fee market in mind. First, it facilitates fee payment in STRK, which is designated as the only native fee token. Second, its bid structure names a price per unit resource, allowing the sequencer can easily compare bids.
```

Comment:
typo: "allowing the sequencer can easily compare bids".

Reply:
```suggestion
Transaction version 3 was introduced with a 1559-type fee market in mind. First, it facilitates fee payment in STRK, which is designated as the only native fee token. Second, its bid structure names a price per unit resource, allowing the sequencer to easily compare bids.
```